allennlp/training/trainer.py

```diff
@@ -0,0 +1,94 @@
+from typing import Any, Dict
+
+import torch
+
+from ..data.dataset import Dataset
+from ..common.params import Params
+from ..common.checks import ConfigurationError
+from .optimizers import get_optimizer_from_params
+from ..data.iterators import Iterator, concrete_iterators
+
+class Trainer:
```

Comment:
Yeah this isn't finished - it will have more features, no fear!

Eventually this will handle:
- Training
- Evaluation on held out data
- Saving and loading a model checkpoint, which will include optimizer state and training metrics
- Model parallism/moving data and models to GPUs.
- Callbacks which effect training, such as saving models, learning rate decay, logging metrics etc.